src/Application.php

```diff
@@ -123,7 +123,7 @@ public function __construct(
      * @throws Exception\BadMethodCallException if the $method is not in $httpRouteMethods.
      * @throws Exception\BadMethodCallException if receiving more or less than 2 arguments.
      */
-    public function __call($method, $args)
+    public function __call($method, array $args)
```

Comment:
This is an interesting case, as, technically, `__call()` is defined originally at the engine level, which allows the user to either include _or omit_ the typehint for the `$args` (or, per the documentation, `$arguments`) argument.

That said, once the method is _declared_ by a class, whatever signature it uses _must_ be used by any extending class.

In this particular case, while we _are_ bumping the major version, it's a bit of a WTF for upgraders if they are overriding the class to add other virtual methods. That said, it's easy enough to correct, as PHP provides the necessary information to update your own code. I'll detail it in the migration notes when merging.